stcrpy/tcr_metrics/tcr_dockq.py

```diff
@@ -0,0 +1,405 @@
+# Adapted from https://github.com/bjornwallner/DockQ
+# Reference: https://doi.org/10.1093/bioinformatics/btae586
+
+# MIT License applies:
+# MIT License
+
+# Copyright (c) 2024 bjornwallner
+
+# Permission is hereby granted, free of charge, to any person obtaining a copy
+# of this software and associated documentation files (the "Software"), to deal
+# in the Software without restriction, including without limitation the rights
+# to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+# copies of the Software, and to permit persons to whom the Software is
+# furnished to do so, subject to the following conditions:
+
+# The above copyright notice and this permission notice shall be included in all
+# copies or substantial portions of the Software.
+
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+# IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+# FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+# AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+# LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+# OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+# SOFTWARE.
+
+import warnings
+from collections import namedtuple
+import itertools
+import Bio
+import json
+from functools import partial
+import logging
+import numpy as np
+import copy
+import os
+
+
+class TCRDockQ:
+    def __init__(self, TCR_pMHC_interface=True, **kwargs):
+        """
+        Initialize the TCRDockQ class. See `Dockq --help` for more information.
+
+        Args:
+            TCR_pMHC_interface (bool): Whether to use evaluate the entire TCR-pMHC interface or evaluate spearate chains. Deaults to True. This will create a merged psudo-chain for the antigen and the TCR.
+            **kwargs: Keyword arguments from DockQ: mapping, small_molecule, capri_peptide, no_align, json, n_cpu, max_chunk, allowed_mismatches, verbose, short, print_results.
+        """
+
+        DockQArgs = namedtuple(
+            "DockQArgs",
+            [
+                "mapping",
+                "small_molecule",
+                "capri_peptide",
+                "no_align",
+                "json",
+                "n_cpu",
+                "max_chunk",
+                "allowed_mismatches",
+                "verbose",
+                "short",
+                "print_results",
+            ],
+        )
+        self.TCR_pMHC_interface = TCR_pMHC_interface
+        self.args = DockQArgs(
+            mapping=kwargs.get("mapping", None),
```

Comment:
Since kwargs is being expanded here anyways, it might be better to set all of these as optional arguments with the defaults already set?